nixos/modules/programs/zsh/zsh.nix

```diff
@@ -187,6 +187,9 @@ in
             fpath=($p/share/zsh/site-functions $p/share/zsh/$ZSH_VERSION/functions $p/share/zsh/vendor-completions $fpath)
         done
 
+        # Alternative method of determining short and full hostname.
+        HOST=`getent hosts 127.0.0.2 | awk '{print $2}'`
```

Comment:
I'm no zsh expert.  Is it suitable to be in zshenv?  It seems it is for interactive use only. If so, it should be in zshrc.

Reply:
.zshenv is used to set important variables
`
.zshenv is sourced on all invocations of the shell, unless the -f option is set. It should contain commands to set the command search path, plus other important environment variables. .zshenv should not contain commands that produce output or assume the shell is attached to a tty.
`

Reply:
`HOST` is used by zsh itself, not by other commands.

After skimming the source code of zsh, it is only used when setting a prompt.  So `HOST` should be set in zshrc.

Reply:
Running two commands on **EVERY** invocation of zsh is bad.

Reply:
Fixed.